jenkins/infra/terraform/main.tf

```diff
@@ -0,0 +1,312 @@
+  }
+}
+
+resource "aws_security_group" "elb" {
+  name        = "ci-elb-sg"
+  description = "Allow inbound traffic from ELB to CI"
+
+  vpc_id = "${data.terraform_remote_state.kubernetes-us-west-2.vpc_id}"
+
+  ingress {
+    from_port   = 80
+    to_port     = 80
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  ingress {
+    from_port   = 443
+    to_port     = 443
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = {
+    Name   = "ci-elb-sg"
+    Region = "${var.region}"
+  }
+}
+
+resource "aws_security_group" "ci" {
+  name        = "ci-sg"
+  description = "Allow inbound traffic to CI from ELB"
+
+  vpc_id = "${data.terraform_remote_state.kubernetes-us-west-2.vpc_id}"
+
+  ingress {
+    from_port = 80
+    to_port   = 80
+    protocol  = "tcp"
+
+    security_groups = [
+      "${aws_security_group.elb.id}",
+    ]
+  }
+
+  ingress {
+    from_port = 443
+    to_port   = 443
+    protocol  = "tcp"
+
+    security_groups = [
+      "${aws_security_group.elb.id}",
+    ]
+  }
+
+  ingress {
+    from_port = 4443
+    to_port   = 4443
+    protocol  = "tcp"
+
+    security_groups = [
+      "${aws_security_group.elb.id}",
+    ]
+  }
+
+  ingress {
+    from_port   = 22
+    to_port     = 22
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = {
+    Name   = "ci-sg"
+    Region = "${var.region}"
+  }
+}
+
+resource "aws_autoscaling_group" "ci" {
+  vpc_zone_identifier = ["${data.aws_subnet_ids.subnet_id.ids}"]
+
+  # This is on purpose, when the LC changes, will force creation of a new ASG
+  name = "ci-${var.project} - ${aws_launch_configuration.ci.name}"
+
+  load_balancers = [
+    "${aws_elb.ci.name}",
+  ]
+
+  lifecycle {
+    create_before_destroy = true
+  }
+
+  max_size                  = "1"
+  min_size                  = "1"
+  desired_capacity          = "1"
+  health_check_grace_period = 1800
+  health_check_type         = "ELB"
+  force_delete              = true
+  launch_configuration      = "${aws_launch_configuration.ci.name}"
+
+  enabled_metrics = [
+    "GroupMinSize",
+    "GroupMaxSize",
+    "GroupDesiredCapacity",
+    "GroupInServiceInstances",
+    "GroupPendingInstances",
+    "GroupStandbyInstances",
+    "GroupTerminatingInstances",
+    "GroupTotalInstances",
+  ]
+
+  tag {
+    key                 = "Name"
+    value               = "ci.${var.region}.${var.domain}"
+    propagate_at_launch = true
+  }
+
+  tag {
+    key                 = "Region"
+    value               = "${var.region}"
+    propagate_at_launch = true
+  }
+}
+
+resource "aws_launch_configuration" "ci" {
+  name_prefix = "ci-${var.project}-"
+
+  image_id = "${data.aws_ami.ubuntu.id}"
+
+  instance_type               = "m5.large"
```

Comment:
I see that the `m5.large` images have only two CPU's? I don't know, but I suspect that won't be enough, even given that MDN is the only client of this Jenkins instance. However, I guess we can increase this as needed?

Reply:
The `m5.large` is the smallest of the latest generation of the General purpose instances, I don't mind bumping this up but just know that your cost will go up. I would prefer to stick to this for now to see if it does cause any issues when builds are going. 

My thinking is since there is really only 2 jobs that this instance size will be ok